src/core/lv2/Lv2Proc.cpp

```diff
@@ -240,11 +243,15 @@ AutomatableModel *Lv2Proc::modelAtPort(const QString &uri)
 
 void Lv2Proc::initPlugin()
 {
+	Lv2Manager* man = Engine::getLv2Manager();
+	m_features.push_back(new LV2_Feature { LV2_URID__map, man->uridMap().mapFeature() });
+	m_features.push_back(new LV2_Feature { LV2_URID__unmap, man->uridMap().unmapFeature() });
```

Comment:
Can these `LV2_Feature`s somehow be kept in smart pointers, or perhaps as non-pointer members of `UridMap`? Obviously a vector of raw pointers is required for `lilv_plugin_instantiate`, but it would be nice to avoid manual memory management.

Reply:
As there's no guarantee that that pointer array can be destroyed after `lilv_plugin_instantiate`, I think we need to keep the `m_features` vector as a member. So the best things I could think of are

1. a "smart" vector of pointers, e.g. a vector that deletes all pointers when it gets destroyed
  -> this is the same as the solution we currently have, just one class deeper
2. ~~using our own smart pointer that guarantees that it only has the pointer as a member~~
  -> IIRC C++ would allow this struct to be larger (due to padding) than the pointer, so that
  does not sound portable
3. a separate structur like `std::vector<some auto pointer type> m_featuresDeleter;`. New pointers will be pushed back to both vectors.

I guess (3) is the best solution. What do you think?

Reply:
Actually I don't think they even need to be individually heap allocated. You could have, for example:
```cpp
std::vector<LV2_Feature> features{
    {LV2_URID__map, man->uridMap().mapFeature()},
    {LV2_URID__unmap, man->uridMap().unmapFeature()},
};
std::vector<const LV2_Feature *> featurePointers;
featurePointers.reserve(features.size());
for(const LV2_Feature &f : features) { featurePointers.push_back(&f); }
```
(Written like this for conciseness - these would be member variables of course.)

Alternatively, `UridMap`'s feature functions could return `LV2_Feature *`s rather than `LV2_URID_Map` and `LV2_URID_Unmap`.
```cpp
class UridMap
{
// ...
    LV2_URID_Map m_uridMap;
    LV2_URID_Unmap m_uridUnmap;
    LV2_Feature m_mapFeature{LV2_URID__map, &m_uridMap};
    LV2_Feature m_unmapFeature{LV2_URID__unmap, &m_uridUnmap};
// ...
    const LV2_Feature *mapFeature() { return &m_mapFeature; }
    const LV2_Feature *unmapFeature() { return &m_unmapFeature; }
// ...
};
//...
void Lv2Proc::initPlugin()
{
    const auto man = Engine::getLv2Manager();
    m_features = {
        man->uridMap().mapFeature(),
        man->uridMap().unmapFeature(),
    };
    //...
}
```

Reply:
I like both of your proposals. The latter one seems to have the disadvantage that you can access the features only as `void*` (right now, you can access them as `LV2_URID_Map`) - though it's currently only used in `void*` context. So I'm more for the former approach.